Source/WebKit/Shared/Cocoa/WebIconUtilities.mm

```diff
@@ -57,46 +61,68 @@ static CGRect squareCropRectForSize(CGSize size)
     return cropRect;
 }
 
-static UIImage *squareImage(CGImageRef image)
+static PlatformImagePtr squareImage(CGImageRef image)
 {
     if (!image)
         return nil;
 
     CGSize imageSize = CGSizeMake(CGImageGetWidth(image), CGImageGetHeight(image));
     if (imageSize.width == imageSize.height)
-        return [UIImage imageWithCGImage:image];
+        return image;
 
     CGRect squareCropRect = squareCropRectForSize(imageSize);
-    RetainPtr<CGImageRef> squareImage = adoptCF(CGImageCreateWithImageInRect(image, squareCropRect));
-    return [UIImage imageWithCGImage:squareImage.get()];
+    return CGImageCreateWithImageInRect(image, squareCropRect);
 }
 
-static RetainPtr<UIImage> thumbnailSizedImageForImage(CGImageRef image)
+static PlatformImagePtr thumbnailSizedImageForImage(CGImageRef image)
 {
-    UIImage *squaredImage = squareImage(image);
+    CGImageRef squaredImage = squareImage(image).get();
```

Comment:
Actually, rather than use `adoptCF` in the return statements, we can do:

`auto squaredImage = adoptCF(squareImage(image));` here.